tasks/reference.js

```diff
@@ -17,108 +17,122 @@ var highlightJs = require('highlightjs');
-                });
-                return definition;
-            });
-            _.forEach(data.paths, function(path, pathUri){
-                _.forEach(path, function(operation,verb){
-                    var operationId = operation.operationId;
-                    var escapeTag = operation.tags[0].replace(/\s/g, '');
-                    if(!data.resources[escapeTag]){
-                        data.resources[escapeTag] = {resourceName: operation.tags[0], operations: {}};
-                    }
-                    var groupedParameters =_.groupBy(operation.parameters, function(parameter){
-                        return parameter.in;
+        var htmlReferencefileName = (version === '1.7') ? 'api-reference-17' : 'api-reference';
+        gulp.src('./content/akeneo-web-api.yaml')
+            .pipe(swagger('akeneo-web-api.json'))
+            .pipe(jsonTransform(function(data, file) {
+                var templateData = data;
+                data.resources = {};
+                data.pimVersion = version;
+                _.map(data.definitions,function(definition){
+                    _.forEach(definition.required, function(requiredProperty){
+                        definition.properties[requiredProperty].required = true;
                     });
-                    _.map(groupedParameters.body, function(parameter){
-                        var readOnlyProperties = [];
-                        _.map(parameter.schema.properties, function(property, propertyName){
-                            property.default = (property.default === 0) ? '0' :
-                                (property.default === null) ? 'null' :
-                                    (property.default === true) ? 'true' :
-                                        (property.default === false) ? 'false' :
-                                            (property.default && _.isEmpty(property.default)) ? '[]' : property.default;
-                            property['x-immutable'] = (verb === 'patch') ? property['x-immutable'] : false;
-                            if(verb === 'post' && property['x-read-only']){
-                                readOnlyProperties.push(propertyName);
-                            }
-                        });
-                        _.forEach(parameter.schema.required, function(requiredProperty){
-                            if(verb !== 'patch'){
-                                parameter.schema.properties[requiredProperty].required = true;
-                            } else {
-                                parameter.schema.properties[requiredProperty].patchRequired = true;
+                    return definition;
+                });
+                _.forEach(data.paths, function(path, pathUri){
+                    _.forEach(path, function(operation,verb){
+                        if((version === '1.7') && (operation['x-versions'][0] === 1.7) || version === '2.0'){
```

Comment:
not missinga `()` ? `(version === '1.7') && (operation['x-versions'][0] === 1.7)` ?